cmd/generator/main.go

```diff
@@ -347,13 +314,10 @@ func getReleasePlugin(release *github.RepositoryRelease, repository *github.Repo
 		}
 
 		if plugin.Manifest.IconPath != "" {
-			iconData, err := getFromTarFile(tar.NewReader(bytes.NewReader(bundleData)), plugin.Manifest.IconPath)
+			plugin.IconData, err = getIconDataFromTarFile(bundleData, plugin.Manifest.IconPath)
```

Comment:
This will end up overwriting `plugin.IconData` if an error occurs -- that /seems/ unexpected from a side effects perspective, IMO.

Reply:
At the end it doesn't make a difference as the code will return anyway next line. But I see you point and changed it.